lib/entities/app-installation.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * UI Extension instances
+ * @namespace AppInstallation
+ */
+
+import { toPlainObject, freezeSys } from 'contentful-sdk-core'
+import cloneDeep from 'lodash/cloneDeep'
+import errorHandler from '../error-handler'
+import enhanceWithMethods from '../enhance-with-methods'
+import {
+  createDeleteEntity
+} from '../instance-actions'
+
+/**
+ * @memberof AppInstallation
+ * @typedef AppInstallation
+ * @prop {Meta.Sys} sys - System metadata
+ * @prop {object} parameters - App Installation specific configuration variables
+ * @prop {function(): Object} toPlainObject() - Returns this App Installation as a plain JS object
+ *
+ */
+
+function createAppInstallationApi (http) {
+  return {
+    /**
+     * Sends an update to the server with any changes made to the object's properties
+     * @memberof AppInstallation
+     * @func update
+     * @return {Promise<AppInstallation>} Object returned from the server with updated changes.
+     * @example
+     * const contentful = require('contentful-management')
+     *
+     * const client = contentful.createClient({
+     *   accessToken: '<content_management_api_key>'
+     * })
+     *
+     * client.getSpace('<space_id>')
+     * .then((space) => space.getEnvironment('<environment_id>'))
+     * .then((environment) => environment.getAppInstallation('<app_definition_id>'))
+     * .then((appInstallation) => {
+     *    appInstallation.parameters.someParameter = 'New Value'
+     *    return appInstallation.update()
+     * })
+     * .then((appInstallation) => console.log(`App installation ${appInstallation.sys.id} was updated`))
+     * .catch(console.error)
+     */
+    update: function () {
```

Comment:
why can't we use `createUpdateEntity` here?

Reply:
Because createUpdateEntity expects an `id` on the root sys object. However the payload of an `appInstallation` does not have an `id` on the `sys` object. It uses the `appDefinition id` which is deeper nested inside the response. So the HTTP call will fail if you use the `createUpdateEntity`

Reply:
@FLoppix